contracts/core/Dispatcher.sol

```diff
@@ -367,7 +373,9 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, ReentrancyGuard, IDi
             )
         );
 
-        delete _portChannelMap[msg.sender][channelId];
+        delete _portChannelMap[msg.sender][channelId]; // Note: We delete the portChannelMap here even on Dapp revert to
+            // avoid having a case where a dapp deployed with a faulty callback cannot close a channel (as is done on
+            // ChannelCloseInit)
```

Comment:
Is this comment referring to IBC-go code?

Reply:
no this comment only refers to on-chain code (either in vibc-core sc or on integrating dapps)

this is an explanation of the tradeoff we made [here](https://github.com/orgs/polymerdao/projects/6/views/7?filterQuery=audit+q2&pane=issue&itemId=59573647)